nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/test/java/org/nuxeo/elasticsearch/test/TestESHintQueryBuilder.java

```diff
@@ -0,0 +1,222 @@
+                GeoDistanceQueryBuilder.NAME);
+        assertTrue(geoDistanceQueryBuilder.isPresent());
+        try {
+            geoDistanceQueryBuilder.get().make(null, ANY_FIELD_NAME, new String[10]);
+            fail("Should raise an IllegalArgumentException");
+        } catch (IllegalArgumentException ia) {
+            assertEquals(
+                    "Hint: org.nuxeo.elasticsearch.hint.GeoDistanceESHintQueryBuilder requires 2 parameters: point and distance",
+                    ia.getMessage());
+        }
+    }
+
+    @Test
+    public void shouldFailWhenMakeGeoShapeESHintQueryWithIllegalArguments() {
+        Optional<ESHintQueryBuilder> geoShapeQueryBuilder = elasticSearchAdmin.getHintByOperator(
+                GeoShapeQueryBuilder.NAME);
+        assertTrue(geoShapeQueryBuilder.isPresent());
+        try {
+            geoShapeQueryBuilder.get().make(null, ANY_FIELD_NAME, new String[10]);
+            fail("Should raise an IllegalArgumentException");
+        } catch (IllegalArgumentException ia) {
+            assertEquals(
+                    "Hint: org.nuxeo.elasticsearch.hint.GeoShapeESHintQueryBuilder requires 4 parameters: shapeId, type, index and path",
+                    ia.getMessage());
+        }
+    }
+
+    /**
+     * {@link org.nuxeo.elasticsearch.hint.RegexESHintQueryBuilder} has a dedicated Test and cannot be integrated in
+     * {@link #shouldEnsureEqualityBetweenESQueriesAndESHintQueries}.
+     * {@link org.nuxeo.elasticsearch.hint.RegexESHintQueryBuilder} is a special case as the operator NXQL
+     * <strong>"regex"</strong> and it's different from {@link org.elasticsearch.index.query.RegexpQueryBuilder#NAME}.
+     * Most of the time the NXQL ESHint and the Elasticsearch operator have the same name. But sometimes they are
+     * different.
+     */
+    @Test
+    public void shouldEnsureEqualityBetweenESRegexQueryAndRegexESHintQueries() {
+        Optional<ESHintQueryBuilder> regexESHintQueryBuilder = elasticSearchAdmin.getHintByOperator("regex");
+        assertTrue(regexESHintQueryBuilder.isPresent());
+        verify(QueryBuilders.regexpQuery(ANY_FIELD_NAME, ANY_VALUE_NAME), regexESHintQueryBuilder.get());
+    }
+
+    protected void verify(AbstractQueryBuilder esQueryBuilder) {
+        Optional<ESHintQueryBuilder> optional = elasticSearchAdmin.getHintByOperator(esQueryBuilder.getWriteableName());
+        assertTrue(optional.isPresent());
+        verify(esQueryBuilder, optional.get());
+    }
+
+    protected void verify(AbstractQueryBuilder esQueryBuilder, ESHintQueryBuilder builder) {
+        EsHint esHint = new EsHint(new EsIdentifierList(ANY_FIELD_NAME), null, null);
+        QueryBuilder esHintQueryBuilder = builder.make(esHint, ANY_FIELD_NAME, ANY_VALUE_NAME);
+
+        assertNotNull(esHintQueryBuilder);
+        ensureEquality(esQueryBuilder, esHintQueryBuilder);
+    }
+
+    protected void ensureEquality(AbstractQueryBuilder esQueryBuilder, QueryBuilder esHintQueryBuilder) {
```

Comment:
Not sure about the need of this method.

Reply:
you mean that not need for this method ; just remove it and replace the call by  
 _assertEquals(esQueryBuilder, esHintQueryBuilder);_
Or
you mean that the test is not need at all ?